lib/kiev/rack/request_logger.rb

```diff
@@ -70,6 +70,15 @@ def log_rack_exception?(exception)
         !Config.instance.ignored_rack_exceptions.include?(exception.class.name)
       end
 
+      def real_ip(request)
+        # use X_FORWARDED_FOR when request comes from load_balancer/proxy
+        if request.env["X_FORWARDED_FOR"]
```

Comment:
`X-Forwarded-For` is (or can be) a list of IPs = all the proxy servers the request has passed through.

so I'd rather go with this approach: https://github.com/rack/rack/blob/master/lib/rack/request.rb#L270-L273

(including rejecting the trusted IPs = local networks etc.)

Reply:
@lipanski  you're totally right and I didn't change that behavior, the `Rack::Request#ip` is still used, see [here](https://github.com/blacklane/kiev/blob/feature/real-ip/lib/kiev/rack/request_logger.rb#L18)

I'm just replacing `REMOTE_ADDR` with `X-Forwarded-For`, look at the new test I've just added https://github.com/blacklane/kiev/pull/6/commits/b759425e5a120bc273f0f8885f998b45b4672530